src/Analyser/MutatingScope.php

```diff
@@ -5008,4 +4987,19 @@ private function getNativeConstantTypes(): array
 		return $constantTypes;
 	}
 
+	/** @return array<string, ExpressionTypeHolder> */
+	private function getSuperglobalExpressionTypes(): array
+	{
+		$superglobalExpressionTypes = [];
+		$exprStrings = ['$GLOBALS', '$_SERVER', '$_GET', '$_POST', '$_FILES', '$_COOKIE', '$_SESSION', '$_REQUEST', '$_ENV'];
+		foreach ($this->expressionTypes as $exprString => $typeHolder) {
```

Comment:
I think we need to handle super globals for native expressions separately.

Reply:
thx, that makes sense I guess, I'm currently trying to implement that, let's see where this goes..

btw., what's the reason that so many scope creations don't pass over native expression? E.g. `enterDeclareStrictTypes()` only retains `$this->expressionTypes`